src/cmd/services/m3coordinator/downsample/metrics_appender.go

```diff
@@ -42,6 +45,11 @@ import (
 	"go.uber.org/zap/zapcore"
 )
 
+var (
+	m3MetricsPrefix              = []byte("__m3")
```

Comment:
As discussed, let's only enable adding type tag and ID re-encoding if and only if rules use special `__m3` tags.